### ISSUE TITLE
Checkout account notice is shown after fields in Checkout block.

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#52619]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In Checkout block, if you're checking logged in or as a guest, you get a small notice that should be below the email, which it is for most cases, but for when there's additional fields, it's shown after them.

### Expected behavior

The guest account notice should be shown alongside the email while the rest are shown in the flow.

![Image](https://github.com/user-attachments/assets/3f47054b-d78c-4283-8102-dfd830f096f9)

![Image](https://github.com/user-attachments/assets/ba7868aa-6d52-4329-b917-fe6ff4c6506a)

### Actual behavior

account creation related things are shown below all fields in contact step

![Image](https://github.com/user-attachments/assets/c5964b00-f065-40bd-a4a2-483f27ba1ac9)

### Steps to reproduce

1. Insert the following code somewhere

```php

add_action(
	'plugins_loaded',
	function () {
		woocommerce_register_additional_checkout_field(
			array(
				'id'       => 'plugin/test-field',
				'label'    => 'Test Field',
				'type'     => 'text',
				'required' => false,
				'location' => 'contact',
			)
		);

		woocommerce_register_additional_checkout_field(
			array(
				'id'       => 'plugin/test-checkbox',
				'label'    => 'Yes, I’d like Woo to send me the latest updates, important announcements, and in-depth release information—all crafted just for developers. I can opt out at anytime.',
				'type'     => 'checkbox',
				'location' => 'contact',
			)
		);
	}
);

```
2. Ensure guest checkout is enabled from WooCommerce -> Settings -> Acoounts and Privacy.
3. Visti Checkout on a guest session.

### WordPress Environment

WP 6.7

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by pierorocca** [comment]
On my test site I get "Create an account with". That's it and no password field.  WIth the new delayed account creation, the "you are checking out as a guest" with a clearly visible Login seems redundant and is noise...and does not lend itself well to WooPay, Link, PayPal Fastlane or whatever else that triggers off of the email address field or that inserts branding into that field.

The During Checkout option here really means "In Checkout". Following the Login link allows account creation from checkout but not in it which to me is the most elegant, conversion friendly option.

**Comment by pierorocca** [comment]
Reviewed and confirmed this is still valid. Marking as ready for engineering.

Current evidence: Open PR #62622 directly closes #52619 and touches the Checkout block contact information/form files involved in the account notice and additional contact-field layout. The issue should be tracked through that PR rather than assigned separately unless the PR is closed without resolving it.

@yjailin I'd also like to get your opinion on the "You are currently checking out out as a guest." message. 

1. Is it necessary?
2. Does it interact well or poorly with things like Stripe Link, PayPal Fastlane, WooPay?
3. The message persists when the user selects "create an account" checkbox which clearly indicates they aren't proceeding as a guest user. I can't reconcile the state conflict.

**Comment by yjailin** [comment]
@pierorocca To your questions:
- The message isn't necessary. The email field itself communicates the state: if it's editable, the shopper is a guest; if it shows a read-only email, they're logged in. A separate message is redundant.
- On payment methods: removing it avoids conflicts with Stripe Link, WooPay, and PayPal Fastlane injecting their own UI near the email field.
- On the state conflict: there's no clean way to reconcile "you are checking out as a guest" with a checked "Create an account" box. Removing the message is the only solution.
Attaching mockups showing what the contact section could look like without the message, in both guest and logged in states.

Guest vs logged:

<img width="993" height="400" alt="Image" src="https://github.com/user-attachments/assets/61584ac5-342c-4053-93e3-8dce78692da5" />

Guest vs logged (with extra fields/checkboxes) 

<img width="949" height="683" alt="Image" src="https://github.com/user-attachments/assets/b4573958-0cbf-4028-beaa-abe4ebace04f" />

cc @elizaan36

**Comment by pierorocca** [comment]
One quirk to reconcile: read-only state. I agree that's the right solution. When we looked at this previously we discovered that the contact email is also being used as the billing email creating the potential for a mismatch between the logged in user account and the email on the purchase. Afaik this is a legacy workaround behavior to satisfy B2B needs of different users placing orders on an account. Any thoughts on how we could maintain backwards compatibility?

**Comment by yjailin** [comment]
@pierorocca ah i didn't tought about the billing email. How about this fpl1YUEIW7suBFO09qymA2-fi-7947_87657?

**Comment by pierorocca** [comment]
Marked as ready for engineering limited to the original scope of the incorrect placement of the notice. @yoannjailin-ai final recommendation is to remove the notice completely?

**Comment by yoannjailin-ai** [comment]
@pierorocca yes, email is pre-filles and log out is display. i think it's fine.
